Project-Hotpot/Project-Hotpot/ViewControllers/HomeViewController.swift

```diff
@@ -102,9 +102,20 @@ class HomeViewController: UIViewController {
     }
     
     func resetSong(){
+        
+        //get the current song that we are going to reset
+        if let curr_song = self.api_instance.lastPlayerState?.track{
+            //add it to the database
+            PFSong.saveSong(song: curr_song)
+        }
+        else{
+            NSLog("last player state wasn't updated properly, is nil")
+            }
```

Comment:
Making another recommendation for liberal use of Cmd+A -> Ctrl+i to auto-fix your indentation.